generators/advanced/index.js

```diff
@@ -0,0 +1,79 @@
+
+const Generator = require('yeoman-generator');
+const chalk = require('chalk');
+const yosay = require('yosay');
+const fs = require('fs');
+const path = require('path');
+
+
+const templatesFolder = path.join(__dirname, 'templates');
+
+module.exports = class extends Generator {
+	prompting() {
+		this.log(
+			yosay(`Welcome to the unreal ${chalk.red('systemic')} generator!`),
+		);
+
+		const prompts = [{
+			type: 'input',
+			name: 'name',
+			message: 'Your service name',
+			default: this.appname.replace(/\s+/g, '-'),
+		},
+		{
+			type: 'input',
+			name: 'description',
+			message: 'A brief description',
+			default: '',
+		},
+		{
+			type: 'input',
+			name: 'author',
+			message: 'The service author',
+			default: 'author',
+		},
+		{
+			type: 'input',
+			name: 'email',
+			message: 'The service author contact email',
+			default: 'author@domain.com',
+		}];
+
+		return this.prompt(prompts).then(props => {
+			this.props = props;
+			this.props.components = ['app', 'config', 'logging', 'express', 'routes', 'store', 'bus', 'controller'];
+		});
+	}
+
+	writing() {
+		const copyFiles = (from, to) => {
+			const configFiles = fs.readdirSync(path.join(templatesFolder, from));
+			configFiles.forEach(file => {
+				this.fs.copyTpl(this.templatePath(`./${from}/${file}`), this.destinationPath(`${to}/${file.replace(/^_/, '')}`), this.props);
+			});
+		};
+
+		const copyAppFiles = () => {
+			this.fs.copy(this.templatePath('./test/.eslintrc'), this.destinationPath('./test/.eslintrc'));
+			this.fs.copy(this.templatePath('./test/*'), this.destinationPath('./test/'));
+			this.fs.copy(this.templatePath('./test/helpers/*'), this.destinationPath('./test/helpers/'));
+			this.props.components.forEach(component => {
+				this.fs.copy(this.templatePath(`./lib/components/${component}/`), this.destinationPath(`./components/${component}/`));
```

Comment:
There is a bug here. `components/app/index.js` should be used as a template as is not a regular file (see `'<%= name %>'`)

Reply:
old but gold bug tracked [here](https://github.com/guidesmiths/generator-systemic/issues/54)